File: SSD.Synopsis.Server.WebAPI/Dtos/GDPRDataDto.cs
using SSD.Synopsis.Server.Core.Models;

namespace SSD.Synopsis.Server.WebAPI.Dtos;

public class GDPRDataDto
{
    public User User { get; set; }
    public IEnumerable<Message> Messages { get; set; }
    public IEnumerable<ChatRoom> ChatRooms { get; set; }
}

File: SSD.Synopsis.Client/src/app/userinfo/gdprdata.dto.ts
import {User} from "../shared/dtos/user.dto";
import {MessageDto} from "../shared/dtos/message.dto";
import {ChatRoomDto} from "../chat/db/db";

export interface GDPRDataDto {
  user: User;
  messages: MessageDto[];
  chatRooms: ChatRoomDto[];
}
